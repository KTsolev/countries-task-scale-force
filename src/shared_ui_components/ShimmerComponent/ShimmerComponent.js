import React from 'react';

export const ShimmerComponent = (props) => {
    const { numberItems=0 } = props;
    let items = [];

    const renderShimmerLines = (number) => {
        for (let i = 1; i <= number; i++) {
            items.push(<div key={i} className='lines shine'></div>);
        }
        return items;
    }

    return (
        <div className='shimmer'>
            {
                renderShimmerLines(numberItems)
            }
        </div>
    );

}