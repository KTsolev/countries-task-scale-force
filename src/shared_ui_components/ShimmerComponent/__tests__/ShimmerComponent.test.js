import React from 'react';
import { ShimmerComponent } from '../ShimmerComponent';
import renderer from 'react-test-renderer';


describe('ShimmerComponent tests', () => {
    it('should render with empty props', () => {
        const tree = renderer.create(<ShimmerComponent />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render 15 tiles', () => {
        const tree = renderer.create(<ShimmerComponent numberItems={15} />);

        expect(tree.root.findAllByProps({className: 'lines shine'}).length).toEqual(15);
        expect(tree.toJSON()).toMatchSnapshot();
    });
})