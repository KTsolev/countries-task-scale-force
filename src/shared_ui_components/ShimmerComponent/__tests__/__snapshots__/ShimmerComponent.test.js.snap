// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShimmerComponent tests should render 15 tiles 1`] = `
<div
  className="shimmer"
>
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
  <div
    className="lines shine"
  />
</div>
`;

exports[`ShimmerComponent tests should render with empty props 1`] = `
<div
  className="shimmer"
/>
`;
